src/components/ScenarioParameters/components/ScenarioParametersInputs/GenericTable.js

```diff
@@ -470,6 +476,51 @@ export const GenericTable = ({
     }
   });
 
+  const onAddRow = useCallback(() => {
+    const newLine = ConfigUtils.createNewTableLine(parameterData.options.columns, parameterData.options.dateFormat);
+    const selectedLines = [
+      ...document.getElementById(`table-${parameterData.id}`).getElementsByClassName('ag-row-selected'),
+    ];
+    let newLineArray = parameter?.agGridRows;
+    if (newLineArray && selectedLines.length > 0) {
+      const lastSelectedLineIndex = selectedLines[selectedLines.length - 1].attributes['row-id'].value;
+      newLineArray.splice(parseInt(lastSelectedLineIndex) + 1, 0, newLine);
+    } else if (newLineArray) newLineArray.unshift(newLine);
+    else newLineArray = [newLine];
+    updateParameterValue({
+      status: UPLOAD_FILE_STATUS_KEY.READY_TO_UPLOAD,
+      tableDataStatus: TABLE_DATA_STATUS.READY,
+      errors: null,
+      uploadPreprocess: { content: _uploadPreprocess },
+      agGridRows: [...newLineArray],
+    });
+  }, [
+    _uploadPreprocess,
+    parameter?.agGridRows,
+    parameterData.id,
+    parameterData.options.columns,
+    parameterData.options.dateFormat,
+    updateParameterValue,
+  ]);
+  const onDeleteRow = useCallback(() => {
+    const selectedLines = document
+      .getElementById(`table-${parameterData.id}`)
+      .getElementsByClassName('ag-row-selected');
+    if (selectedLines.length > 0) {
+      const newRows = parameter.agGridRows;
+      for (let i = 0; selectedLines[i]; i++) {
+        newRows[selectedLines[i].attributes['row-id'].value] = undefined;
+      }
+      updateParameterValue({
+        status: UPLOAD_FILE_STATUS_KEY.READY_TO_UPLOAD,
+        tableDataStatus: TABLE_DATA_STATUS.READY,
+        errors: null,
+        uploadPreprocess: { content: _uploadPreprocess },
+        agGridRows: newRows.filter((row) => row),
```

Comment:
is this way to delete rows specifically recommended by ag-grid ? I'm not sure that creating one loop to assign undefined and then filter undefined values is the best way to filter an array.

Reply:
If we use the AgGrid system, the webapp will no longer be synchronised. Have has result a table who won't be saved

Reply:
the question is not about synchronizing the table, it's about your function that filters rows : it is looping through `newRows` assigning `undefined` to every row to delete and than filtering all `undefined` values, I think we can try to find a way to filter it directly.